crates/host/src/workload_identity.rs

```diff
@@ -0,0 +1,35 @@
+use anyhow::{Context as _, Result};
+
+// TODO(joonas): Figure out better naming here
+const AUTH_SERVICE_AUDIENCE_ENV: &str = "WASMCLOUD_WORKLOAD_IDENTITY_AUTH_SERVICE_AUDIENCE";
+
+/// WorkloadIdentityConfig is used by the experimental workload-identity feature
+#[derive(Clone, Default, Debug)]
+pub struct WorkloadIdentityConfig {
+    /// auth_service_audience represents the value expected by the Auth Callout Service,
+    /// typically this should look something like "spiffe://wasmcloud.dev/auth-callout"
+    pub auth_service_audience: String,
+}
+
+impl WorkloadIdentityConfig {
+    /// Fetch workload identity configuration from environment variables
+    #[cfg(unix)]
+    pub fn from_env() -> Result<Self> {
+        // TODO(joonas): figure out better naming here. maybe this should be interpolated from a trust domain?
+        // This needs to follow format like: "spiffe://{spiffe_trust_domain}/{nats_auth_callout_service}"
```

Comment:
Do these naming comments need to be addressed?

Reply:
Not yet, this was a comment for future.